cmake/Modules/FindComputeCpp.cmake

```diff
@@ -230,6 +230,7 @@ function(__build_ir)
   # using the same source file will be generated with a different rule.
   set(baseSyclName ${CMAKE_CURRENT_BINARY_DIR}/${SDK_BUILD_IR_TARGET}_${sourceFileName})
   set(outputSyclFile ${baseSyclName}.sycl)
+  set(outputDeviceFile ${baseSyclName}.o)
```

Comment:
Does this output an object file now, or is this a bitcode file? FindComputeCpp.cmake doesn't use -sycl-driver yet (my fault, I never really tried hard to add it in a sensible way)

Reply:
This output the kernel in the requested format, so it will be bitcode for SPIR or the binary blob for SPIRV etc. The .o is just used as a generic extension.

Reply:
Ah, I see. I was slightly confused by the .o suffix, so could we maybe switch to .bc? Or we could also use the value of COMPUTECPP_BITCODE, which would at least indicate what the file contains, even if it's not the "real" file extension. Let me know your thoughts :)

Reply:
No matter what we pick it is going to be confusing, it should be .s for ptx, .bc for SPIR and .spv for SPIRV...

Reply:
I was thinking about this and you could do something like:
```CMake
set(IR_MAP_spir64 bc)
set(IR_MAP_ptx64 s)
# later...
set(outputDeviceFile ${baseSyclName}.${IR_MAP_${COMPUTECPP_BITCODE}})
```

Kind of nasty, but would get accurate file names every time!

Reply:
That looks nasty to me, but if this can avoid confusion why not

Reply:
I tried my hand at adding it, and I think putting it in its own file is the easiest option for now - I prepared a patch if that helps:
```diff
diff --git a/cmake/Modules/ComputeCppIRMap.cmake b/cmake/Modules/ComputeCppIRMap.cmake
new file mode 100644
index 0000000..b3dd770
--- /dev/null
+++ b/cmake/Modules/ComputeCppIRMap.cmake
@@ -0,0 +1,17 @@
+cmake_minimum_required(VERSION 3.4.3)
+
+# These should match the types of IR output by compute++
+set(IR_MAP_spir bc)
+set(IR_MAP_spir64 bc)
+set(IR_MAP_spir32 bc)
+set(IR_MAP_spirv spv)
+set(IR_MAP_spirv64 spv)
+set(IR_MAP_spirv32 spv)
+set(IR_MAP_aorta-x86_64 o)
+set(IR_MAP_aorta-aarch64 o)
+set(IR_MAP_custom-spir64 bc)
+set(IR_MAP_custom-spir32 bc)
+set(IR_MAP_custom-spirv64 spv)
+set(IR_MAP_custom-spirv32 spv)
+set(IR_MAP_ptx64 s)
+set(IR_MAP_amdgcn s)
diff --git a/cmake/Modules/FindComputeCpp.cmake b/cmake/Modules/FindComputeCpp.cmake
index 66d6538..98809bb 100644
--- a/cmake/Modules/FindComputeCpp.cmake
+++ b/cmake/Modules/FindComputeCpp.cmake
@@ -31,6 +31,7 @@
 
 cmake_minimum_required(VERSION 3.4.3)
 include(FindPackageHandleStandardArgs)
+include(ComputeCppIRMap)
 
 set(COMPUTECPP_USER_FLAGS "" CACHE STRING "User flags for compute++")
 separate_arguments(COMPUTECPP_USER_FLAGS)
@@ -230,7 +231,7 @@ function(__build_ir)
   # using the same source file will be generated with a different rule.
   set(baseSyclName ${CMAKE_CURRENT_BINARY_DIR}/${SDK_BUILD_IR_TARGET}_${sourceFileName})
   set(outputSyclFile ${baseSyclName}.sycl)
-  set(outputDeviceFile ${baseSyclName}.o)
+  set(outputDeviceFile ${baseSyclName}.${IR_MAP_${COMPUTECPP_BITCODE}})
   set(depFileName ${baseSyclName}.sycl.d)
 
   set(include_directories "$<TARGET_PROPERTY:${SDK_BUILD_IR_TARGET},INCLUDE_DIRECTORIES>")
```
I took the list of IR formats from `compute++` listing what the acceptable options to `-sycl-target` are.